packaging/package-indexer/indexer/rpm_indexer.py

```diff
@@ -0,0 +1,304 @@
+
+    def __cache_gpg_key_passphase_if_needed(self, gnupghome_dir: Path) -> None:
+        if self.__gpg_key_passphrase is None:
+            return
+
+        with NamedTemporaryFile(mode="w+", encoding="utf-8") as f:
+            command = [
+                "gpg",
+                "--armor",
+                "--detach-sign",
+                "--sign",
+                f.name,
+            ]
+            command = self.__add_gpg_security_params(command)
+            command = self.__add_gpg_passphrase_params_if_needed(command)
+            env = {"GNUPGHOME": str(gnupghome_dir)}
+
+            self._log_info("Caching GPG key passphrase.", gnupghome_dir=str(gnupghome_dir))
+            utils.execute_command(command, env=env, input=self.__gpg_key_passphrase)
+
+    def __stop_gpg_agent_if_needed(self, gnupghome: TemporaryDirectory) -> None:
+        if self.__gpg_key_passphrase is None:
+            return
+
+        command = ["gpgconf", "--kill", "gpg-agent"]
+        env = {"GNUPGHOME": str(gnupghome.name)}
+
+        self._log_info("Stopping GPG agent.", gnupghome=gnupghome.name)
+        utils.execute_command(command, env=env)
+
+
+class StableRpmIndexer(RpmIndexer):
+    def __init__(
+        self,
+        incoming_remote_storage_synchronizer: RemoteStorageSynchronizer,
+        indexed_remote_storage_synchronizer: RemoteStorageSynchronizer,
+        run_id: str,
+        cdn: CDN,
+        gpg_key_path: Path,
+        gpg_key_passphrase: Optional[str],
+    ) -> None:
+        super().__init__(
+            incoming_remote_storage_synchronizer=incoming_remote_storage_synchronizer,
+            indexed_remote_storage_synchronizer=indexed_remote_storage_synchronizer,
+            incoming_sub_dir=Path("stable", run_id, "rpm"),
+            dist_dir=Path("stable"),
+            cdn=cdn,
+            gpg_key_path=gpg_key_path,
+            gpg_key_passphrase=gpg_key_passphrase,
+        )
+
+
+class NightlyRpmIndexer(RpmIndexer):
+    PKGS_TO_KEEP = 10
+
+    def __init__(
+        self,
+        incoming_remote_storage_synchronizer: RemoteStorageSynchronizer,
+        indexed_remote_storage_synchronizer: RemoteStorageSynchronizer,
+        cdn: CDN,
+        run_id: str,
+        gpg_key_path: Path,
+        gpg_key_passphrase: Optional[str],
+    ) -> None:
+        super().__init__(
+            incoming_remote_storage_synchronizer=incoming_remote_storage_synchronizer,
+            indexed_remote_storage_synchronizer=indexed_remote_storage_synchronizer,
+            incoming_sub_dir=Path("nightly", run_id, "rpm"),
+            dist_dir=Path("nightly"),
+            cdn=cdn,
+            gpg_key_path=gpg_key_path,
+            gpg_key_passphrase=gpg_key_passphrase,
+        )
+
+    def __get_pkg_timestamps_in_dir(self, dir: Path) -> List[str]:
+        timestamp_regexp = re.compile(r"\+([^_]+)_")
+        pkg_timestamps: List[str] = []
+
+        for deb_file in dir.rglob("syslog-ng-devel*.rpm"):
```

Comment:
axosyslog?

Reply:
Thanks, nice catch.

An example file name (0.8.1 comes from not having tags in my fork):
```
axosyslog-devel-0.8.1.18105.g4ca9958-2+20241117T182608.x86_64.rpm
```

The pattern is:
```
\+([^.]+).
```

Will match like:
![image](https://github.com/user-attachments/assets/616cbdf0-69e5-418a-99a2-73ea56fb9a81)